src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordPropertySymbol.cs

```diff
@@ -46,71 +57,39 @@ protected override Location TypeLocation
         protected override SyntaxTokenList GetModifierTokens(SyntaxNode syntax)
             => new SyntaxTokenList();
 
-        protected override ArrowExpressionClauseSyntax? GetArrowExpression(SyntaxNode syntax)
-            => null;
-
-        protected override bool HasInitializer(SyntaxNode syntax)
-            => true; // Synthesized record properties always have a synthesized initializer
-
         public override SyntaxList<AttributeListSyntax> AttributeDeclarationSyntaxList
             => new SyntaxList<AttributeListSyntax>();
 
-        protected override void GetAccessorDeclarations(
-            CSharpSyntaxNode syntax,
-            DiagnosticBag diagnostics,
-            out bool isAutoProperty,
-            out bool hasAccessorList,
-            out bool accessorsHaveImplementation,
-            out bool isInitOnly,
-            out CSharpSyntaxNode? getSyntax,
-            out CSharpSyntaxNode? setSyntax)
-        {
-            isAutoProperty = true;
-            hasAccessorList = false;
-            getSyntax = setSyntax = syntax;
-            isInitOnly = true;
-            accessorsHaveImplementation = false;
-        }
-
         protected override void CheckForBlockAndExpressionBody(CSharpSyntaxNode syntax, DiagnosticBag diagnostics)
         {
             // Nothing to do here
         }
 
-        protected override DeclarationModifiers MakeModifiers(
-            SyntaxTokenList modifiers,
-            bool isExplicitInterfaceImplementation,
-            bool isIndexer,
-            bool accessorsHaveImplementation,
-            Location location,
-            DiagnosticBag diagnostics,
-            out bool modifierErrors)
-        {
-            Debug.Assert(!isExplicitInterfaceImplementation);
-            Debug.Assert(!isIndexer);
-            modifierErrors = false;
-
-            return DeclarationModifiers.Public;
-        }
-
         protected override SourcePropertyAccessorSymbol CreateAccessorSymbol(
             bool isGet,
             CSharpSyntaxNode? syntax,
             PropertySymbol? explicitlyImplementedPropertyOpt,
-            string aliasQualifierOpt,
+            string? aliasQualifierOpt,
             bool isAutoPropertyAccessor,
             bool isExplicitInterfaceImplementation,
             DiagnosticBag diagnostics)
         {
             Debug.Assert(syntax is object);
             Debug.Assert(isAutoPropertyAccessor);
+
+            var overriddenProperty = OverriddenProperty;
+            var overriddenAccessor = overriddenProperty is null ?
+                null :
+                (isGet ? overriddenProperty.GetMethod : overriddenProperty.SetMethod);
+            string name = overriddenAccessor?.Name ??
+                SourcePropertyAccessorSymbol.GetAccessorName(_sourceName, isGet, this.IsCompilationOutputWinMdObj());
```

Comment:
This looks suspicious. Doesn't SourcePropertyAccessorSymbol.CreateAccessorSymbol takes care of this?

Reply:
It looks like SourcePropertyAccessorSymbol constructor deals with this.

```
            if (this.IsOverride)
            {
                MethodSymbol overriddenMethod = this.OverriddenMethod;
                if ((object)overriddenMethod != null)
                {
                    // If this accessor is overriding a method from metadata, it is possible that
                    // the name of the overridden method doesn't follow the C# get_X/set_X pattern.
                    // We should copy the name so that the runtime will recognize this as an override.
                    _name = overriddenMethod.Name;
                }
            }
```

The ```SourcePropertyAccessorSymbol.GetAccessorName``` is also a code duplication.

I think we can remove all of it.

---
In reply to: [446523951](https://github.com/dotnet/roslyn/pull/45336#discussion_r446523951) [](ancestors = 446523951)